From 97d2ccfbfdbaa9d19620117fdcc4e278e44a3469 Mon Sep 17 00:00:00 2001
From: Ryan Schmidt
Date: Wed, 10 Feb 2021 23:54:59 -0600
Subject: [PATCH 09/14] Use correct syntax for the -dylib_file flag

Use correct syntax for the -dylib_file flag (no space after colon).
Fixes configure failure on macOS Big Sur and later.

configure: error: GL library could not be found, please specify its location with --with-gl-lib.

config.log contains:

configure:17551: /usr/bin/clang -o conftest -pipe -Os -DGL_SILENCE_DEPRECATION -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.0.sdk -arch x86_64  -L/opt/local/lib -Wl,-headerpad_max_install_names -Wl,-syslibroot,/Library/Developer/CommandLineTools/SDKs/MacOSX11.0.sdk -a
clang: error: no such file or directory: '/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib'

---
 m4/gl.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/m4/gl.m4 b/m4/gl.m4
index 09c84ca7bdb9..a4c789f873b6 100644
--- a/m4/gl.m4
+++ b/m4/gl.m4
@@ -48,11 +48,11 @@ FRAMEWORK_OPENGL=""
 PRELIBS="$LIBS"
 LIBS="$LIBS -Xlinker -framework -Xlinker OpenGL"
 # -Xlinker is used because libtool is busted prior to 1.6 wrt frameworks
 AC_TRY_LINK([#include <OpenGL/gl.h>], [glBegin(GL_POINTS)],
     [GL_DYLIB="/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib"
-     FRAMEWORK_OPENGL="-Xlinker -framework -Xlinker OpenGL -dylib_file $GL_DYLIB: $GL_DYLIB"
+     FRAMEWORK_OPENGL="-Xlinker -framework -Xlinker OpenGL -dylib_file $GL_DYLIB:$GL_DYLIB"
      ac_cv_search_glBegin="$FRAMEWORK_OPENGL"
      AC_MSG_RESULT(yes)],
     [AC_MSG_RESULT(no)])
 if test x"$FRAMEWORK_OPENGL" != "x"; then
   with_gl_lib="$FRAMEWORK_OPENGL"
--
2.47.1

